app/src/main/java/com/duckduckgo/app/browser/omnibar/experiments/FadeOmnibarLayout.kt

```diff
@@ -127,12 +126,22 @@ class FadeOmnibarLayout @JvmOverloads constructor(
     }
 
     override fun render(viewState: ViewState) {
-        val experimentalViewState = viewState.copy(
-            showBrowserMenu = false,
-            showFireIcon = false,
-            showTabsMenu = false,
-        )
-        super.render(experimentalViewState)
+        if (viewState.viewMode is ViewMode.CustomTab) {
+            // adds a drop shadow for the AppBarLayout, in case it was removed at any point
+            outlineProvider = ViewOutlineProvider.BACKGROUND
+            super.render(viewState)
+        } else {
+            // removes the drop shadow from the AppBarLayout to make it appear flat in the view hierarchy
+            outlineProvider = null
```

Comment:
this shouldn’t be needed every single time the view state is rendered. the omnibar won’t change from custom tab to browser mode.

Reply:
> the omnibar won’t change from custom tab to browser mode

While that's true now, I don't want to bake that assumption into the codebase. The layout doesn't know it, and it should just react to what's in the view model.

This setter function is guarded by an equality check under the hood, so there's no perf penalty.